lib/ansible/modules/cloud/scaleway/scaleway_images_facts.py

```diff
@@ -0,0 +1,177 @@
+      - ARM64-128GB
+      - C1
+      - C2S
+      - C2M
+      - C2L
+      - START1-XS
+      - START1-S
+      - START1-M
+      - START1-L
+      - X64-15GB
+      - X64-30GB
+      - X64-60GB
+      - X64-120GB
+
+  timeout:
+    description:
+    - Timeout for API calls
+    required: false
+    default: 30
+'''
+
+EXAMPLES = '''
+- name: Get image information
+  scaleway_images_facts:
+    name: Ubuntu Bionic
+    region: par1
+    commercial_type: START1
+'''
+
+RETURN = '''
+data:
+    description: This is only present when C(state=present)
+    returned: when C(state=present)
+    type: dict
+    sample: {
+        "images": [
+            {
+                "arch": "x86_64",
+                "creation_date": "2018-07-26T13:00:02.726274+00:00",
+                "default_bootscript": {
+                    "architecture": "x86_64",
+                    "bootcmdargs": "LINUX_COMMON scaleway boot=local nbd.max_part=16",
+                    "default": false,
+                    "dtb": "",
+                    "id": "54ee2857-8ffb-4323-abba-964f55fea4a2",
+                    "initrd": "http://169.254.42.24/initrd/initrd-Linux-x86_64-v3.14.5.gz",
+                    "kernel": "http://169.254.42.24/kernel/x86_64-mainline-lts-4.9-4.9.93-rev1/vmlinuz-4.9.93",
+                    "organization": "11111111-1111-4111-8111-111111111111",
+                    "public": true,
+                    "title": "x86_64 mainline 4.9.93 rev1"
+                },
+                "extra_volumes": [],
+                "from_server": null,
+                "id": "67dcb730-fc17-421c-a328-f67438c74bd6",
+                "modification_date": "2018-07-26T13:00:02.726274+00:00",
+                "name": "Ubuntu Bionic Beaver",
+                "organization": "51b656e3-4865-41e8-adbc-0c45bdd780db",
+                "public": true,
+                "root_volume": {
+                    "id": "2fef7fe5-7201-487d-9cc1-058b2e6e9a12",
+                    "name": "snapshot-ea04f81d-b0bd-499f-b23a-887d01f14eb6-2018-07-25_21:28",
+                    "size": 50000000000,
+                    "volume_type": "l_ssd"
+                },
+                "state": "available"
+            }
+        ]
+    }
+'''
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.module_utils.scaleway import ScalewayAPI, SCALEWAY_LOCATION, SCALEWAY_COMMERCIAL_TYPES_ARCH_MAPPING
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+
+
+def core(module):
+    api_token = module.params['oauth_token']
+    region = module.params["region"]
+    name = module.params["name"]
+    arch = SCALEWAY_COMMERCIAL_TYPES_ARCH_MAPPING[module.params["commercial_type"]]
+
+    compute_api = ScalewayAPI(module=module,
+                              headers={'X-Auth-Token': api_token},
+                              base_url=SCALEWAY_LOCATION[region]["api_endpoint"])
+
+    response = compute_api.get("images?%s" % urlencode({"arch": arch, "name": name}))
```

Comment:
When `name` parameter isn't specified by user, then this query returns always an empty list.

Reply:
I've added name as a required parameter